core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2020 The Hyve B.V.
+# This code is licensed under the GNU Affero General Public License (AGPL),
+# version 3, or (at your option) any later version.
+#
+
+#
+# This file is part of cBioPortal.
+#
+# cBioPortal is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""OncoKB annotation import script for discrete CNA files.
+"""
+
+import argparse
+import importlib
+import logging.handlers
+import os
+import requests
+import sys
+from os import path
+from pathlib import Path
+
+# configure relative imports if running as a script; see PEP 366
+# it might passed as empty string by certain tooling to mark a top level module
+if __name__ == "__main__" and (__package__ is None or __package__ == ''):
+    # replace the script's location in the Python search path by the main
+    # scripts/ folder, above it, so that the importer package folder is in
+    # scope and *not* directly in sys.path; see PEP 395
+    sys.path[0] = str(Path(sys.path[0]).resolve().parent)
+    __package__ = 'importer'
+    # explicitly import the package, which is needed on CPython 3.4 because it
+    # doesn't include https://github.com/python/cpython/pull/2639
+    importlib.import_module(__package__)
+
+from . import cbioportal_common
+from . import libImportOncokb
+from . import validateData
+
+required_cna_columns = ['Hugo_Symbol']
+
+
+# from: cbioportal-frontend file CopyNumberUtils.ts
+cna_alteration_types = {
+    "DELETION": -2,
+    "LOSS": -1,
+    "GAIN": 1,
+    "AMPLIFICATION": 2,
+}
+
+
+def main_import(args):
+
+    study_dir = args.study_directory
+    server_url = args.url_server
+
+    # get a logger to emit messages
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    exit_status_handler = validateData.MaxLevelTrackingHandler()
+    logger.addHandler(exit_status_handler)
+
+    # set default message handler
+    text_handler = logging.StreamHandler(sys.stdout)
+    text_handler.setFormatter(
+        cbioportal_common.LogfileStyleFormatter(study_dir))
+    collapsing_text_handler = cbioportal_common.CollapsingLogMessageHandler(
+        capacity=5e5,
+        flushLevel=logging.CRITICAL,
+        target=text_handler)
+    collapsing_text_handler.setLevel(logging.INFO)
+    logger.addHandler(collapsing_text_handler)
```

Comment:
This logger initiation script is duplicated, is it possible to merge together?

Reply:
No, this is not possible. It is boiler-plate code for setting the environment. Sharing is difficult because we are not dealing with a class based architecture.